src/main/webapp/components/lists/lists.js

```diff
@@ -97,58 +120,103 @@ const Lists = props => {
           </ListTitle>
         )
 
-        return (
+        return list['list_bookmarks'] == undefined ? (
           <IndexCardItem
             key={list.id}
             title={<Title />}
-            subHeader={<ListInfo listSize={list['list_bookmarks'].length} />}
+            subHeader={<div>Empty</div>}
             onClick={() => {
               handleClick(list)
             }}
           >
             <Actions>
-              <EditAction />
               <DeleteAction />
             </Actions>
           </IndexCardItem>
+        ) : (
+          <React.Fragment>
+            <IndexCardItem
+              key={list.id}
+              title={<Title />}
+              subHeader={<ListInfo listSize={list['list_bookmarks'].length} />}
+              onClick={() => {
+                handleClick(list)
+              }}
+            >
+              <Actions>
+                <EditAction onEdit={handleEditorOpen} />
+                <DeleteAction itemName="list" isWritable={true} />
+              </Actions>
+            </IndexCardItem>
+            <ListCreatePopover
+              key={index}
+              list={list}
+              anchorEl={editorAnchorEl}
+              onClose={handleEditorClose}
+              open={isEditorOpen}
+              setList={setList}
+              onSave={onSave}
+            />
+          </React.Fragment>
         )
       })
     : []
 
-  return (
-    lists && (
-      <React.Fragment>
-        <div style={{ display: 'flex' }}>
-          {selected
-            ? ListCards.filter(list => list.key === selected)[0]
-            : ListCards[0]}
-          <Button
-            color="primary"
-            variant="contained"
-            style={{ marginTop: 20, marginBottom: 20, marginRight: 20 }}
-            onClick={handleOpen}
-          >
-            <ExpandMoreIcon />
-          </Button>
-        </div>
-
-        <Popover
-          open={open}
-          anchorEl={anchorEl}
-          onClose={handleClose}
-          anchorOrigin={{
-            vertical: 'top',
-            horizontal: 'left',
-          }}
-          transformOrigin={{
-            vertical: 'top',
-            horizontal: 'left',
-          }}
+  return lists ? (
+    <React.Fragment>
+      <div style={{ display: 'flex' }}>
+        {selected
+          ? ListCards.filter(list => list.props.children[0].key === selected)[0]
+          : ListCards[0].props.children}
```

Comment:
I think it would be nice if we could achieve this logic without using children props. It seems like you could move the key from the IndexCardItem to the React.Fragment on lines 137-138

:question: Why return the ListCards element directly in the first case, but return the props.children in the second case?

Reply:
Each ListCard has both the list and the pop over as children since they are both wrapped in a fragment. I'm still returning the ListCard on the first one too. It's just using the key from the list to determine if it's selected.